basic/line.go

```diff
@@ -14,7 +14,7 @@ func (Line) String() string { return "Line" }
 // NewLine creates a line given pairs for floats.
 func NewLine(pointPairs ...float64) *Line {
 	line := new(Line)
-	if (2 % len(pointPairs)) != 0 {
+	if (len(pointPairs) % 2) != 0 {
 		panic("NewLine requires pair of points.")
```

Comment:
That's a good question. This panic has saved me. And the `2 % len(poinPairs)` is bug that has slipped by us. Could change it to return an error, and I am inclined to lean that way.